adabot/arduino_libraries.py

```diff
@@ -162,6 +172,27 @@ def validate_release_state(repo):
 
     return
 
+def validate_travis(repo):
+    """Validate if a repo has .travis.yml.
+    """
+    repo_has_travis = github.get("/repos/" + repo["full_name"] + "/contents/.travis.yml")
+    if repo_has_travis.ok:
+        return True
+
+def has_ino(repo):
+    """Validate if a repo has any *.ino files.
+    """
+    repo_has_ino = github.get("/search/code?q=extension:ino+repo:adafruit/" + repo["name"])
+    if repo_has_ino.ok and repo_has_ino.json()["total_count"] > 0:
+        return True
+
+def has_library_properties(repo):
```

Comment:
You could conceivably avoid this function altogether. `validate_library_properties()` above simply bails out of the function if `library.properties` doesn't exist in the repo. Tweaking that would do the same, and save another call to GitHub API. Its line # 111 in your commit.

Original line: https://github.com/adafruit/adabot/blob/master/adabot/arduino_libraries.py#L101